### PR TITLE
Cipher

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

File: src/cipher/cipher.h [hunk 1]
```diff
@@ -23,15 +23,17 @@
  * @param[out] must have inlen + 64 bytes.
  * @param[inout] out_len must be inlen + 64 incoming, and will be the actual
  * length of the output.
+ * @param[in] secret 32 bytes secret
```
Comment (NickeZ) [inline src/cipher/cipher.h]: document the whole interface

File: src/cipher/cipher.h [hunk 2]
```diff
@@ -23,15 +23,17 @@
...
     const uint8_t* in,
-    int in_len,
+    size_t in_len,
     uint8_t* out,
-    int* out_len,
+    size_t* out_len,
     const uint8_t* secret);
 
 /**
+ * @param[in] must have at least 64 bytes.
```
Comment (NickeZ) [inline src/cipher/cipher.h]: document the whole interface

File: src/cipher/cipher.h (hunk 1 above)
Comment (benma) [reply src/cipher/cipher.h]: Done

File: src/cipher/cipher.h (hunk 2 above)
Comment (benma) [reply src/cipher/cipher.h]: Done

**Pull request merged**